kernel/ksu.c

```diff
@@ -244,6 +244,16 @@ static int handler_pre(struct kprobe *p, struct pt_regs *regs)
 		return 0;
 	}
 
+	// Both root manager and root processes should be allowed to get version
+	if( __manager_uid == current_uid().val || 0 == current_uid().val ) {
```

Comment:
Check the arg2 first, and then check if it is valid role.
And also, please reuse the `is_manager`, don't repeat yourself.